### PR TITLE
update styling for whole site

### PR DESCRIPTION
**Pull request by Holquaria**

### CODE REVIEW

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Jamie, nice login functionality, but have a few things here to refactor:

- Please display the delete button on the posted comment straight away once it's been posted, so the user doesn't have to reload the whole app to delete the comment.
- Half of the error message is covered by other elements. (Please see screenshot below)
- Please display an instant feedback when a comment is deleted successfully.
- Your textarea's value can be reset to an empty string, If you wanted to display a text like 'Have your say...", use the 'placeholder' attribute for the 'textarea' tag. Also need to disable resizing of the text area. 

Please amend your code and send us a quick PR with it.

Screenshot 2023-01-12 at 10 40 34

File: src/components/Articles.jsx [hunk 1]
```diff
@@ -28,16 +28,12 @@ export const Articles = () => {
...
-                <br />
-                Comments: {article.comment_count}</p>
-                Votes: {article.votes}
-                
+                <div className='article-title'><Link to={`/articles/${article.article_id}`}><h3>{article.title}</h3></Link></div>
+                <p className='article-author'>Author: {article.author}</p>
+                <p className='article-date'>Date posted: {article.created_at.substring(11, 16)}, {article.created_at.substring(0, 10)}</p>
+                <p className='comment-count'>Comments: {article.comment_count}</p>
+                <p className='vote-count'>Votes: {article.votes}</p>
+                <p className='topic'>Topic: {(article.topic.slice(0,1)).toUpperCase()}{article.topic.substring(1)}</p>
```
Comment (Poonam-raj) [inline src/components/Articles.jsx]: Mkes a lot of sense to extract this into a articlecard component to de-bloat the component

File: src/components/CommentArray.jsx [hunk 2]
```diff
@@ -1,12 +1,27 @@
...
             return <li key={comment.comment_id} className='comment-card'>
-                <p className="username">{comment.author}</p>
+                <div className="comment-header"><p className="username">{comment.author}</p>
                 <p className="date">
                 Date posted: {comment.created_at.substring(11, 16)}, {comment.created_at.substring(0, 10)}
-                </p>
+                </p></div>
                 <p className="comment-body">{comment.body}</p>
+                {comment.author === 'tickle122' && loggedIn === true ? <button disabled={deletingComment === true} onClick={() => {removeComment(comment.comment_id)}} className="delete-comment">Delete</button> : null}
             </li>
```
Comment (Poonam-raj) [inline src/components/CommentArray.jsx]: Again makes sense to extract this into a comment card component

File: src/components/CommentError.jsx [hunk 3]
```diff
@@ -1,10 +1,12 @@
-export const CommentError = (retryPostComment, postedComment) => {
+export const CommentError = ({retryPostComment, postingComment}) => {
+  console.log(postingComment)
```
Comment (Poonam-raj) [inline src/components/CommentError.jsx:2]: remove console logs before submitting

File: src/components/CommentArray.jsx [hunk 4]
```diff
@@ -1,12 +1,27 @@
...
+    const [deletingComment, setDeletingComment] = useState(null)
+
+    const removeComment = (comment_id) => {
+        setDeletingComment(true)
+        deleteComment(comment_id).then(() => {
+            setCommentRemoved((currCount) => {
+                return currCount + 1
+            })
+        })
+    }
```
Comment (Poonam-raj) [inline src/components/CommentArray.jsx]: if this functionality is almost identical to the removeComment situation in NewComment then you have a good case for extracting this button out to a delete comment component to make it reusable

File: src/components/Comments.jsx [hunk 5]
```diff
@@ -19,12 +20,12 @@ export const Comments = ({loggedIn}) => {
        }).catch((err) => {
         setLoading('error')
        })
-      }, []);
+      }, [commentRemoved]);
```
Comment (Poonam-raj) [inline src/components/Comments.jsx]: I don't want you to re-render the comments after removal - I want you to filter it locally instead

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
Some stuff I want you to do before you merge - please make those changes and let me know any questions - I'd be happy for you to make a PR again just to once over the changes

**Review by Pipe-Smoking-Rabbit — APPROVED** [review]
This is a massive styling update that's really brought the look of everything together! 
I only have some small feedback points relating to the deleting comments functionality:
- When in offline mode, trying to delete a comment results in an infinite loading stage where the users sees `"Deleting comment..."`, instead try to handle this error and give some user friendly feedback that something has gone wrong when the request fails
- I would advise refactoring the logged in user to make use of context ( it seems to be just a state currently ), it's good to show off to potential employers that you have experience with context!

**Pull request merged**